### ISSUE TITLE
[17.3.4 - 17.5.0] can not clone or push to repository

### ISSUE THREAD

**Issue by Basti-Fantasti** [sameersbn/docker-gitlab#3031]
Hi all,

I just updated my gitlab instance today to 17.5.0 from 17.3.4 -> 17.4.2 -> 17.5.0

- I was able to add a new repository using the web interface
- I could check this newly created repository out to my PC
- Then I added a gitignore file, which I was also able to push
- but now I added some source code files and wanted to push them and now I only receive this:

I set debug in the docker-compose file to true to get more information.

```shell
$ git push
remote:
remote: ========================================================================
remote:
remote: Internal API unreachable
remote:
remote: ========================================================================
remote:
fatal: Could not read from remote repository.

Please make sure you have the correct access rights
and the repository exists.
```

But the repo exists, as I said I was able to clone it and push a single file once...

When now trying to git push one more time the git push command just hangs there...

EDIT 1:
Without any issues I can clone the existing repo in a new folder and have the one file in it, which I just pushed to the empty repository.

Also adding a submodule fails.
It starts enumerating but only creates the empty directory.

**Comment by Basti-Fantasti** [comment]
This is strange... Now I reverted my gitlab back to 17.3.4 and it's not working there, too.

I need to talk to my colleagues next week to check if it's working on their projects.
I've tried on two different PCs now with the same result.

I tried on 
- 17.3.4
- 17.3.5
- 17.5.0

all show the same problem.

The clone process just hangs here:

```shell
remote: Enumerating objects: 9239, done.
remote: Counting objects: 100% (270/270), done.
remote: Compressing objects: 100% (267/267), done.

```

**Comment by Basti-Fantasti** [comment]
I also checked all the log files but found no hint which could guide my further Investigations Info any specific direction

**Comment by Basti-Fantasti** [comment]
I figured it out. It may be of interest to other Windows users as well. 
The issue was related to a buggy git version

The git 2.47.0 version has this bug built in in connection with gitlab instances.

I fixed it as suggested [here](https://gitlab.com/gitlab-org/gitlab/-/issues/499350)
by manually updating git over the command line to `2.47.0.windows.2`

by typing 

```shell
git update-git-for-windows
```

This fixed it for me.
